### PR TITLE
[DatadogMonitor] add CRD

### PR DESCRIPTION
**Pull request by celenechang**
### What does this PR do?

Adds `DatadogMonitor` CRD

### CODE REVIEW

File: api/v1alpha1/datadogmonitor_types.go [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+	// Title is the monitor title 
+	Title string `json:"title,omitempty"`
+	// Message is the message to include in a monitor notification
+	Message string `json:"message,omitempty"`
+	// Tags is the monitor tags used to organize monitors
+	Tags []string `json:"tags,omitempty"`
+	
+	// Options are the optional parameters of a monitor
+	Options DatadogMonitorOptions `json:"options,omitempty"`
+	}
```
Comment (clamoriniere) [inline api/v1alpha1/datadogmonitor_types.go:28]:
```suggestion
}
```

**Review by clamoriniere — COMMENTED** [review]
Looks great 👍 

we can think later about: auto downtime configuration feature.

File: config/crd/kustomization.yaml [hunk 2]
```diff
@@ -4,19 +4,22 @@
 resources:
 - bases/v1beta1/datadoghq.com_datadogagents.yaml
 - bases/v1beta1/datadoghq.com_datadogmetrics.yaml
+- bases/datadoghq.com_datadogmonitors.yaml
```
Comment (clamoriniere) [inline config/crd/kustomization.yaml:7]: should we have also `v1beta1` in the path too?

Comment (celenechang) [reply config/crd/kustomization.yaml:7]: looks like it, thank you

File: config/samples/kustomization.yaml [hunk 3]
```diff
@@ -2,4 +2,5 @@
 resources:
 - datadog-agent-all.yaml
 - datadogmetric-v1alpha1.yaml
+- datadoghq_v1alpha1_datadogmonitor.yaml
```
Comment (celenechang) [inline config/samples/kustomization.yaml:5]: this is not named in the same format as the others (although it matches the filename); should i change it?

File: controllers/suite_test.go [hunk 4]
```diff
@@ -65,6 +65,9 @@ var _ = BeforeSuite(func(done Done) {
 	err = apiregistrationv1.AddToScheme(scheme.Scheme)
 	Expect(err).NotTo(HaveOccurred())
 
+	err = datadoghqv1alpha1.AddToScheme(scheme.Scheme)
```
Comment (vboulineau) [inline controllers/suite_test.go:68]: Duplicated

File: config/samples/kustomization.yaml (hunk 3 above)
Comment (celenechang) [reply config/samples/kustomization.yaml:5]: talked to vincent about this, going to leave it as is since it does not matter too much.

**Review by CharlyF — APPROVED** [review]
sounds good

File: controllers/datadogmonitor_controller.go [hunk 5]
```diff
@@ -0,0 +1,42 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2019 Datadog, Inc.
+
+package controllers
+
+// import (
```
Comment (celenechang) [inline controllers/datadogmonitor_controller.go:8]: This file was autogenerated; implementation to be worked on in a separate PR

**Review by clamoriniere — APPROVED** [review]
👍

**Comment by celenechang** [comment]
Closing due to git issues encountered when signing old commits
